pandas/core/indexes/datetimes.py

```diff
@@ -389,27 +380,6 @@ def tz(self, value):
         raise AttributeError("Cannot directly set timezone. Use tz_localize() "
                              "or tz_convert() as appropriate")
 
-    @property
-    def size(self):
-        # TODO: Remove this when we have a DatetimeTZArray
-        # Necessary to avoid recursion error since DTI._values is a DTI
-        # for TZ-aware
-        return self._ndarray_values.size
```

Comment:
> Why are you removing those?  Those will need to be added back

Because I am OK with needing to add them back in a few days (hopefully)

Reply:
But can you then try to explain me what the advantage is of moving it now?

Reply:
1) To make it clear what still needs to be moved/implemented at the Array level.  e.g. Tom's PeriodArray PR implements some things in PeriodArray that should instead be in DatetimeLikeArrayMixin.  Moving these prevents this kind of mixup.

2) Because there are already a bunch of things that are going to need to be inherited from `self.values`, its better to get them all in one place and do that all at once.

3) Because in the next pass I'll be implementing a decorator to do something like:

```
# TODO: enable this decorator once Datetime/Timedelta/PeriodIndex .values
#   points to a pandas ExtensionArray
# @inherit_from_values(["ndim", "shape", "size", "nbytes",
#                       "asi8", "freq", "freqstr"])
class DatetimeIndexOpsMixin(DatetimeLikeArrayMixin):
```

Reply:
> Moving these prevents this kind of mixup.

As long as one of the index classes is still inheriting from the ArrayMixin, there will be wrong / strange mixups, that need to be cleaned up

> Because in the next pass I'll be implementing a decorator to do something like:

But how would you do that if the underlying values don't yet have those attributes, because it is not yet our internal array class? 

And why not move them *when* implementing such a decorator? Then you actually have overview of the full changes.

Reply:
You have sufficiently frustrated me into reverting this so we can move this down the field.